financial-services-accelerator/components/org.wso2.financial.services.accelerator.common/src/main/java/org/wso2/financial/services/accelerator/common/util/HTTPClientUtils.java

```diff
@@ -59,60 +59,114 @@ public class HTTPClientUtils {
+
+    private HTTPClientUtils() {
+        // Prevent instantiation
+    }
 
     /**
-     * Get closeable https client.
+     * Initialize the connection manager for HTTPS protocol.
      *
-     * @return Closeable https client
+     * @param maxTotal     Maximum pool connections
+     * @param maxPerRoute Maximum connections per route
      * @throws FinancialServicesException FinancialServicesException exception
      */
-    @Generated(message = "Ignoring since method contains no logics")
-    public static CloseableHttpClient getHttpsClient() throws FinancialServicesException {
+    private static void initConnectionManagerForHttpsProtocol(int maxTotal, int maxPerRoute)
+            throws FinancialServicesException {
 
-        SSLConnectionSocketFactory sslsf = createSSLConnectionSocketFactory();
+        if (connectionManager == null) {
+            synchronized (HTTPClientUtils.class) {
+                if (connectionManager == null) {
+                    SSLConnectionSocketFactory sslsf = createSSLConnectionSocketFactory();
 
-        Registry<ConnectionSocketFactory> socketFactoryRegistry = RegistryBuilder.<ConnectionSocketFactory>create()
-                .register(HTTP_PROTOCOL, new PlainConnectionSocketFactory())
-                .register(HTTPS_PROTOCOL, sslsf)
-                .build();
+                    Registry<ConnectionSocketFactory> socketFactoryRegistry = RegistryBuilder
+                            .<ConnectionSocketFactory>create()
+                            .register(HTTP_PROTOCOL, new PlainConnectionSocketFactory())
+                            .register(HTTPS_PROTOCOL, sslsf)
+                            .build();
 
-        final PoolingHttpClientConnectionManager connectionManager = (socketFactoryRegistry != null) ?
-                new PoolingHttpClientConnectionManager(socketFactoryRegistry) :
-                new PoolingHttpClientConnectionManager();
+                    connectionManager = new PoolingHttpClientConnectionManager(socketFactoryRegistry);
 
-        // configuring default maximum connections
-        connectionManager.setMaxTotal(FinancialServicesConfigParser.getInstance().getConnectionPoolMaxConnections());
-        connectionManager.setDefaultMaxPerRoute(FinancialServicesConfigParser.getInstance()
-                .getConnectionPoolMaxConnectionsPerRoute());
+                    connectionManager.setMaxTotal(maxTotal);
+                    connectionManager.setDefaultMaxPerRoute(maxPerRoute);
+                }
+            }
+        }
+    }
 
-        return HttpClients.custom().setConnectionManager(connectionManager).build();
+    /**
+     * Initialize the connection manager for HTTP protocol.
+     *
+     * @param maxTotal     Maximum pool connections
+     * @param maxPerRoute Maximum connections per route
+     */
+    private static void initConnectionManagerForHttpProtocol(int maxTotal, int maxPerRoute) {
+
+        if (connectionManager == null) {
```

Comment:
can't we use a singleton connection manager and httpClient,instead volatile and follow initialization-on-demand holder idiom design pattern. That way you don't need to add syncronized code block on each method..

Reply:
Fixed in https://github.com/wso2/financial-services-accelerator/pull/514/commits/c65da8dcad96670acbfc4a7aa8bfdeb4b53ba1da